### ISSUE TITLE
What are the newest versions of poppler and fontforge that will build with this project?

### ISSUE THREAD

**Issue by asinning** [pdf2htmlEX/pdf2htmlEX#15]
I'm trying to understand the meaning of the two previous issues.  Is there a branch that I can checkout that will compile with newer versions of poppler and fontforge?

What are the newest versions that folks have successfully been able to compile with?  Thanks!

**Comment by Rockstar04** [comment]
I have a script I use for building pdf2htmlEX that uses Poppler 62, and font forge, git branch 20170731.

I start with my fork, which is a mix of other people's patches: https://github.com/Rockstar04/pdf2htmlEX

**Comment by asinning** [comment]
Thanks Rockstar04!  Is your script in your fork? Would you be willing to share? I would be very grateful.

**Comment by Rockstar04** [comment]
https://gist.github.com/Rockstar04/c77f9f46f15be7b156aaed9a34bb5188

**Comment by asinning** [comment]
Awesome work Rockstar04! I have a comment and a related question. 

Comment: Compiling poppler with the DENABLE_LIBOPENJPEG=none flag (line 56) produced poor results when converting pdfs with many layers of images.  We found that some layers where missing.
```
cmake -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=/usr -DENABLE_XPDF_HEADERS=ON -DENABLE_LIBOPENJPEG=none
```

In order to compile without the DENABLE_LIBOPENJPEG flag, I did need to install the following dependencies:
```
apt-get install libgirepository1.0-dev
apt-get install libopenjp2-7-dev
apt-get install libgtk-3-dev
```

Question: What is the purpose of the DENABLE_LIBOPENJPEG flag?

**Comment by Rockstar04** [comment]
From what I assume it disabled popplers ability to import or export JPEGs? For the steps compiling Font Forge and Poppler, you can look to their official documentation for more information about compiling them from source.

**Comment by brecke** [comment]
Thanks a lot for the script. I'll try adapting it to run in alpine linux so I can use that as a docker image. Have you tried it? There's [this image](https://hub.docker.com/r/bwits/pdf2htmlex-alpine/~/dockerfile/) but it's a bit outdated (alpine 3.2).

**Comment by brecke** [comment]
...and here it is, in case you want it: https://hub.docker.com/r/oaeproject/oae-pdf2htmlex-docker/ It's using alpine:3.8

Source is [here](https://github.com/oaeproject/oae-pdf2htmlEX-docker).

Feel free to use this as a base for someone else's work and/or include it in the wiki. Thanks a lot for your hard work!

**Comment by jgoldfar** [comment]
Quick work on that alpine image! I had taken a look at reducing the image size using alpine as a base image, but ended up building an image using Debian: https://hub.docker.com/r/jgoldfar/pdf2htmlex-stable/
Looks like an interesting application you all are working on!

**Comment by amit777** [comment]
Just an FYI. It looks like a lot of security updates have been made to Poppler since .63 (https://poppler.freedesktop.org/releases.html).   I'm going to try getting this working on centos7 building from source. Will post the steps if I get it working.

**Comment by brecke** [comment]
@amit777 feel free to fork my repo (alpine based) above and please report back your findings

**Comment by amit777** [comment]
i was able to get it compiled with .63 on centos7.. had to do some updates to the .sh script.  I was a little too ambitious and tried to get it working with poppler .72.  Which required me  (i think to upgrade C++ environment to GCC7).. I made it further, but then got other compliation errors.  I'm no where near knowledgable enough to solve them.

Anyway, I'm seeing a bunch of forks and not sure which one is the best to work of.. 
pdf2htmlEX/pdf2htmlEX
Rockstar
Alpine (which i don't know what that is).

**Comment by brecke** [comment]
@amit777 alpine is a linux distro very lightweight and therefore common when building docker images

The docker image I published above is built on alpine, and compiles pdf2htmlEX succesfuly, you can check versions in the Dockerfile.

**Comment by amit777** [comment]
@brecke, so i've never really worked with Docker before but started with yours.. Got it working pretty well.  I noticed a small difference in library versions between your docker dev build and the centos7 build I have.  Below are the --versions of each.  libfontforge and cairo are the version differences.  

-- version output on Centos 7 build
```
pdf2htmlEX version 0.15.0
Copyright 2012-2015 Lu Wang <coolwanglu@gmail.com> and other contributors
Libraries: 
  poppler 0.63.0
  libfontforge 20190219
  cairo 1.15.12
Default data-dir: /usr/local/share/pdf2htmlEX
Supported image format: png jpg svg

```

--version output on your docker build:
```
pdf2htmlEX version 0.15.0
Copyright 2012-2015 Lu Wang <coolwanglu@gmail.com> and other contributors
Libraries: 
  poppler 0.63.0
  libfontforge 20181011
  cairo 1.14.8
Default data-dir: /build/usr/share/pdf2htmlEX
Supported image format: png jpg svg

```

**Comment by brecke** [comment]
Yeah, well, I didn't really care for having the most recent versions of all dependencies, once I got one working that was just about enough 🤷🏻‍♂️ feel free to try different versions though and fork at will
